components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ServiceTest.java

```diff
@@ -172,4 +213,174 @@ public void testIdentityOAuth2Exception() throws Exception {
+//        assertFalse(oAuth2Service.revokeTokenByOAuthClient(revokeRequestDTO).isError());
+
+        when(OAuth2Util.authenticateClient(anyString(), anyString())).thenThrow(new InvalidOAuthClientException(" "));
+        assertEquals(oAuth2Service.revokeTokenByOAuthClient(revokeRequestDTO).getErrorMsg(), "Unauthorized Client");
+    }
+
+    @Test
+    public void testIdentityOAuth2ExceptionForRevokeTokenByOAuthClient() throws Exception {
+
+        when(oAuthEventInterceptorProxy.isEnabled()).thenReturn(true);
+        doThrow(new IdentityOAuth2Exception("")).when(oAuthEventInterceptorProxy)
+                .onPreTokenRevocationByClient(any(OAuthRevocationRequestDTO.class), anyMap());
+        when(oAuthComponentServiceHolder.getOAuthEventInterceptorProxy()).thenReturn(oAuthEventInterceptorProxy);
+        mockStatic(OAuthComponentServiceHolder.class);
+        when(OAuthComponentServiceHolder.getInstance()).thenReturn(oAuthComponentServiceHolder);
+
+        mockStatic(OAuthServerConfiguration.class);
+        when(oAuthServerConfiguration.getPersistenceProcessor()).thenReturn(persistenceProcessor);
+        TokenMgtDAO tokenMgtDAO = mock(TokenMgtDAO.class);    //mock methods for tokenMgtDAO
+        whenNew(TokenMgtDAO.class).withAnyArguments().thenReturn(tokenMgtDAO);
+
+        OAuthRevocationRequestDTO revokeRequestDTO = new OAuthRevocationRequestDTO();
+        revokeRequestDTO.setConsumerKey("testCosumerKey");
+        revokeRequestDTO.setToken("testToken");
+        String result = "Error occurred while revoking authorization grant for applications";
+        assertEquals(oAuth2Service.revokeTokenByOAuthClient(revokeRequestDTO).getErrorMsg(), result);
+    }
+
+
+    public void setUpRevokeToken() throws Exception {
+
+        when(oAuthEventInterceptorProxy.isEnabled()).thenReturn(true);
+        doNothing().when(oAuthEventInterceptorProxy).onPostTokenRevocationByClient
+                (any(OAuthRevocationRequestDTO.class), any(OAuthRevocationResponseDTO.class), any(AccessTokenDO.class),
+                        any(RefreshTokenValidationDataDO.class), any(HashMap.class));
+
+        when(oAuthComponentServiceHolder.getOAuthEventInterceptorProxy()).thenReturn(oAuthEventInterceptorProxy);
+        mockStatic(OAuthComponentServiceHolder.class);
+        when(OAuthComponentServiceHolder.getInstance()).thenReturn(oAuthComponentServiceHolder);
+
+        TokenPersistenceProcessor persistenceProcessor = new PlainTextPersistenceProcessor();
+
+        when(authenticatedUser.toString()).thenReturn("testAuthenticatedUser");
+
+        mockStatic(OAuthServerConfiguration.class);
+        when(oAuthServerConfiguration.getPersistenceProcessor()).thenReturn(persistenceProcessor);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(oAuthServerConfiguration);
+        when(oAuthServerConfiguration.getTimeStampSkewInSeconds()).thenReturn(3600L);
+        when(oAuthServerConfiguration.isRevokeResponseHeadersEnabled()).thenReturn(true);
```

Comment:
It would be nice if you create a private method for mocking for OAuth2Util so that you can avoid repeating the same procedure in other tests as well. Seems like you have the same procedure in **testIsPKCESupportEnabled** as well.

Reply:
okay